src/extension.ts

```diff
@@ -255,6 +255,28 @@ export class Extension implements vscode.Disposable {
     return this.enabled = enabled
   }
 
+  /**
+   * Make all selections in the editor non-empty by selecting at least one character.
+   */
+  normalizeSelections(editor: vscode.TextEditor) {
+    if (this.modeMap.get(editor.document) !== Mode.Normal) { return; }
+    if (this.configuration.get('selections.allowEmpty')) { return; }
```

Comment:
This call may be expensive, instead please use `observePreference` in the `constructor`.

Reply:
Done.